### ISSUE TITLE
False positive for `implicit-saturating-sub` with `else` block

### ISSUE THREAD

**Issue by narpfel** [rust-lang/rust-clippy#7831]
Lint name: `clippy::implicit-saturating-sub`

I tried this code:

```rust
fn foo(mut i: u32) {
    if i > 0 {
        i -= 1;
    }
    else {
        println!("side effect");
    }
}

fn main() {
    foo(42);
}
```

I expected to see this happen: This should not trigger `implicit-saturating-sub` because the `else` block performs side effects.

Instead, this happened:
```console
$ cargo clippy -- -D clippy::implicit-saturating-sub
error: implicitly performing saturating subtraction
 --> src/main.rs:2:5
  |
2 | /     if i > 0 {
3 | |         i -= 1;
4 | |     }
5 | |     else {
6 | |         println!("side effect");
7 | |     }
  | |_____^ help: try: `i = i.saturating_sub(1);`
  |
  = note: requested on the command line with `-D clippy::implicit-saturating-sub`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#implicit_saturating_sub
```

Additionally, as the lint is `MachineApplicable`, `cargo clippy --fix` changes the behaviour:
```rust
fn foo(mut i: u32) {
    i = i.saturating_sub(1);
}

fn main() {
    foo(42);
}
```

If I understand the code correctly, the error is in https://github.com/rust-lang/rust-clippy/blob/d50cfd26e5fe72afef8dd60adda42faddfe45d91/clippy_lints/src/implicit_saturating_sub.rs#L46, which was changed in b97d4c062b4d99139f14960daa7ce99d63625833 from requiring `None` for the `else` block to matching `..`:
https://github.com/rust-lang/rust-clippy/commit/b97d4c062b4d99139f14960daa7ce99d63625833#diff-32d04babbaeb1ddc0077df6f5f7dae0661db4c876180bc8711196c16cfdaa59dL45-R46

### Meta

**Rust version (`rustc -Vv`):**
```
rustc 1.57.0-nightly (4e89811b4 2021-10-16)
binary: rustc
commit-hash: 4e89811b46323f432544f9c4006e40d5e5d7663f
commit-date: 2021-10-16
host: x86_64-unknown-linux-gnu
release: 1.57.0-nightly
LLVM version: 13.0.0
```

```console
$ cargo clippy -- --version
clippy 0.1.57 (4e89811 2021-10-16)
```
